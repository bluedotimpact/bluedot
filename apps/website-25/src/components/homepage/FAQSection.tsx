import { Collapsible, Section } from '@bluedot/ui';

const FAQSection = () => {
  return (
    <Section className="faq-section" title="Frequently asked questions">
      <Collapsible title="How can I maximise my chances of getting accepted onto the courses?">
        <p>Check you are the <a href="https://bluedot.org/blog/am-i-the-right-fit-for-bluedots-courses/">right fit</a> and read our tips for making a <a href="https://aisafetyfundamentals.com/blog/avoid-governance-application-mistakes/">great application.</a></p>
      </Collapsible>
      <Collapsible title="I’d like to run your course for my work/university/ local AIS group… can you help with training and materials?">
        <p>We are excited that you want to <a href="https://bluedot.org/running-versions-of-our-courses/">run a version of our courses</a> and will support you in any way we can. A good place to start is by looking through our <a href="https://course.bluedot.org/home/facilitator-training">facilitator training course</a> curriculum.</p>
      </Collapsible>
      <Collapsible title="Who funds your work?">
        <p>BlueDot Impact is primarily funded by <a href="https://www.openphilanthropy.org/">Open Philanthropy</a>, and is a non-profit based in the UK (company number <a href="https://find-and-update.company-information.service.gov.uk/company/14964572">14964572</a>). Our courses are free but upon course completion we give our participants the opportunity to <a href="https://donate.stripe.com/5kA3fpgjpdJv6o89AA">Support Us</a>.</p>
      </Collapsible>
      <Collapsible title="How can I collaborate/partner with BlueDot Impact on AI safety initiatives?">
        <p>We love to hear from potential collaborators so please <a href="https://bluedot.org/contact/">reach out</a>.</p>
      </Collapsible>
      <Collapsible title="How can I stay informed about BlueDot Impact’s latest roles?">
        <p>Please see more information about our job openings <a href="/careers">here</a> including an <a href="https://airtable.com/appB9rItpTpWkNqEI/shrulKbUxr7NhgnKJ">expression of interest form</a>.</p>
      </Collapsible>
    </Section>
  );
};

export default FAQSection;
