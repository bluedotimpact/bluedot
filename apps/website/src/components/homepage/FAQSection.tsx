import { Collapsible, Section } from '@bluedot/ui';
import { A, P } from '../Text';

const FAQSection = () => {
  return (
    <Section className="faq-section !overflow-visible" title="Frequently asked questions">
      <Collapsible className="faq-section__item" title="What is AI Safety?">
        <P>AI Safety is the field working to ensure powerful AI systems benefit rather than harm humanity. As AI systems become more capable of affecting our world, we need to solve key challenges: How do we make AI systems reliably do what we want? How do we govern their development? How do we define what "beneficial" means?</P>
        <P>You can read more about these <A href="https://aisafetyfundamentals.com/blog/what-is-ai-alignment/">different aspects of AI safety here</A> and the <A href="https://aisafetyfundamentals.com/blog/ai-risks/">range of AI risks here</A>.</P>
      </Collapsible>
      <Collapsible className="faq-section__item" title="How can I maximise my chances of getting accepted onto the courses?">
        <P>Check you are the <A href="https://bluedot.org/blog/am-i-the-right-fit-for-bluedots-courses/">right fit</A> and read our tips for making a <A href="https://aisafetyfundamentals.com/blog/avoid-governance-application-mistakes/">great application.</A></P>
      </Collapsible>
      <Collapsible className="faq-section__item" title="I’d like to run your course with my work/university/group of friends. Can you help with training and materials?">
        <P>We’d love for you to do this! All our reading and discussion materials are freely available online. You are welcome to run a local version of our course so long as you <A href="https://bluedot.org/running-versions-of-our-courses/">follow our guidelines</A>.
        </P>
        <P>If you want tips for your discussion facilitation, we recommend taking a look through our <A href="https://course.bluedot.org/home/facilitator-training">facilitator training course</A>.</P>
      </Collapsible>
    </Section>
  );
};

export default FAQSection;
