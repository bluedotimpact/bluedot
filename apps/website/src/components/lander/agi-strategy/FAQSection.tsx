import { useState } from 'react';

const FAQSection = () => {
  const [openQuestions, setOpenQuestions] = useState<string[]>([]);

  const faqData = [
    {
      id: 'funding',
      question: 'Can I just apply for funding?',
      answer: 'Funding is only available for graduates of the course.',
    },
    {
      id: 'bluedot',
      question: 'Who is BlueDot Impact?',
      answer: (
        <>
          We're a London-based startup. Since 2022, we've trained 5,000 people, with ~1,000 now working on making AI go well.
          <br /><br />
          Our courses are the main entry point into the AI safety field.
          <br /><br />
          We're an intense 4-person team. We've raised $35M in total, including $25M in 2025.
        </>
      ),
    },
  ];

  const handleToggle = (id: string) => {
    setOpenQuestions((prev) => {
      return prev.includes(id)
        ? prev.filter((questionId) => questionId !== id) // Remove if already open
        : [...prev, id]; // Add if not open
    });
  };

  return (
    <section className="w-full bg-white">
      <div className="max-w-max-width mx-auto px-5 min-[680px]:px-8 lg:px-spacing-x py-12 min-[680px]:py-16 lg:pt-24 lg:pb-20">
        <div className="max-w-[928px] mx-auto flex flex-col gap-12 md:gap-16">
          <h2 className="text-[28px] min-[680px]:text-[32px] xl:text-[36px] font-semibold leading-[125%] tracking-[-0.01em] text-[#13132E] text-center">
            Frequently Asked Questions
          </h2>

          <div className="flex flex-col gap-6">
            {faqData.map((item) => {
              const isOpen = openQuestions.includes(item.id);

              return (
                <div
                  key={item.id}
                  className="border border-[rgba(19,19,46,0.1)] bg-white rounded-xl overflow-hidden"
                >
                  <button
                    type="button"
                    onClick={() => handleToggle(item.id)}
                    className={`w-full flex items-center gap-8 text-left px-8 cursor-pointer transition-all duration-300 ease ${
                      isOpen ? 'pt-6 pb-2' : 'py-6'
                    } ${!isOpen ? 'hover:bg-gray-50' : ''}`}
                    aria-expanded={isOpen}
                    aria-controls={`faq-answer-${item.id}`}
                  >
                    <span className="text-[18px] font-semibold leading-[125%] text-[#13132E] flex-grow">
                      {item.question}
                    </span>
                    <svg
                      width="16"
                      height="17"
                      viewBox="0 0 16 17"
                      fill="none"
                      xmlns="http://www.w3.org/2000/svg"
                      className={`flex-shrink-0 transition-transform duration-300 ease ${
                        isOpen ? 'rotate-45' : ''
                      }`}
                    >
                      <path d="M0 8.5H16M8 0.5L8 16.5" stroke="#001133" strokeWidth="2" />
                    </svg>
                  </button>

                  {/* Answer container with animation */}
                  <div
                    id={`faq-answer-${item.id}`}
                    className={`grid transition-[grid-template-rows] duration-300 ease ${
                      isOpen ? 'grid-rows-[1fr]' : 'grid-rows-[0fr]'
                    }`}
                  >
                    <div className="overflow-hidden">
                      <div className="px-8 pb-6">
                        <div className="text-[18px] font-normal leading-[160%] text-[#13132E] opacity-80">
                          {item.answer}
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              );
            })}
          </div>
        </div>
      </div>
    </section>
  );
};

export default FAQSection;
