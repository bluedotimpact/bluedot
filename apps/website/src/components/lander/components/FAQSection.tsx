import { useState } from 'react';

/**
 * Represents a single FAQ item with question and answer
 */
export type FAQItem = {
  /** Unique identifier for this FAQ item (used for accordion state management) */
  id: string;
  /** The question text */
  question: string;
  /** The answer text or React element */
  answer: React.ReactNode;
};

/**
 * Props for the FAQSection component
 */
export type FAQSectionProps = {
  /** Section heading displayed at the top */
  title: string;
  /** Array of FAQ items to display */
  items: FAQItem[];
};

const FAQSection = ({ title, items }: FAQSectionProps) => {
  const [openQuestions, setOpenQuestions] = useState<string[]>([]);

  const handleToggle = (id: string) => {
    setOpenQuestions((prev) => {
      return prev.includes(id)
        ? prev.filter((questionId) => questionId !== id) // Remove if already open
        : [...prev, id]; // Add if not open
    });
  };

  return (
    <section className="w-full bg-white">
      <div className="max-w-max-width mx-auto px-5 min-[680px]:px-8 lg:px-spacing-x py-12 min-[680px]:py-16 lg:pt-24 lg:pb-20">
        <div className="max-w-[928px] mx-auto flex flex-col gap-12 md:gap-16">
          <h2 className="text-[28px] min-[680px]:text-[32px] xl:text-[36px] font-semibold leading-[125%] tracking-[-0.01em] text-[#13132E] text-center">
            {title}
          </h2>

          <div className="flex flex-col gap-6">
            {items.map((item) => {
              const isOpen = openQuestions.includes(item.id);

              return (
                <div
                  key={item.id}
                  className="border border-[rgba(19,19,46,0.1)] bg-white rounded-xl overflow-hidden"
                >
                  <button
                    type="button"
                    onClick={() => handleToggle(item.id)}
                    className={`w-full flex items-center gap-8 text-left px-8 cursor-pointer transition-all duration-300 ease ${
                      isOpen ? 'pt-6 pb-2' : 'py-6'
                    } ${!isOpen ? 'hover:bg-gray-50' : ''}`}
                    aria-expanded={isOpen}
                    aria-controls={`faq-answer-${item.id}`}
                  >
                    <span className="text-[18px] font-semibold leading-[125%] text-[#13132E] flex-grow">
                      {item.question}
                    </span>
                    <svg
                      width="16"
                      height="17"
                      viewBox="0 0 16 17"
                      fill="none"
                      xmlns="http://www.w3.org/2000/svg"
                      className={`flex-shrink-0 transition-transform duration-300 ease ${
                        isOpen ? 'rotate-45' : ''
                      }`}
                    >
                      <path d="M0 8.5H16M8 0.5L8 16.5" stroke="#001133" strokeWidth="2" />
                    </svg>
                  </button>

                  {/* Answer container with animation */}
                  <div
                    id={`faq-answer-${item.id}`}
                    className={`grid transition-[grid-template-rows] duration-300 ease ${
                      isOpen ? 'grid-rows-[1fr]' : 'grid-rows-[0fr]'
                    }`}
                  >
                    <div className="overflow-hidden">
                      <div className="px-8 pb-6">
                        <div className="text-[18px] font-normal leading-[160%] text-[#13132E] opacity-80">
                          {item.answer}
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              );
            })}
          </div>
        </div>
      </div>
    </section>
  );
};

export default FAQSection;
