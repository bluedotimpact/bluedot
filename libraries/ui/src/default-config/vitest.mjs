/* eslint-disable import/no-extraneous-dependencies */
import { defineConfig, defaultExclude } from 'vitest/config';
import react from '@vitejs/plugin-react';
import * as dotenv from 'dotenv';

/**
 * @param {import('vitest/config').UserConfig} [config] 
 * @returns {import('vitest/config').UserConfig}
 */
export const withDefaultBlueDotVitestConfig = async (config) => defineConfig({
  ...config,
  plugins: [react(), ...(config?.plugins ?? [])],
  test: {
    environment: 'happy-dom',
    env: dotenv.config({ path: '.env.test' }).parsed,

    ...config?.test,

    exclude: [...defaultExclude, '**/.{turbo,next}/**', ...(config?.test?.exclude ?? [])],
  },
});
