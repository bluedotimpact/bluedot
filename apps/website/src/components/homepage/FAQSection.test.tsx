import { render } from '@testing-library/react';
import { describe, expect, test } from 'vitest';
import FAQSection from './FAQSection';

describe('FAQSection', () => {
  test('renders default as expected', () => {
    const { container } = render(<FAQSection />);
    expect(container).toMatchSnapshot();
  });
});
