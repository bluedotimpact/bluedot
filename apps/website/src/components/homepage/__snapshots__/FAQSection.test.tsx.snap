// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FAQSection > renders default as expected 1`] = `
<div>
  <section
    class="section section-body faq-section !overflow-visible"
  >
    <div
      class="section-heading__title-container flex justify-between items-center gap-space-between mb-6"
    >
      <div
        class="section-heading__content flex-1 flex flex-col gap-2"
      >
        <h2
          class="section-heading__title relative ml-[-0.08em] bluedot-h2"
        >
          Frequently asked questions
        </h2>
      </div>
    </div>
    <div
      class="section__body"
    >
      <details
        class="collapsible max-w-max-width border-b border-color-divider py-4 last:border-b-0 group marker:hidden [&_summary::-webkit-details-marker]:hidden faq-section__item"
      >
        <summary
          class="collapsible__header flex justify-between w-full cursor-pointer py-6 text-left"
        >
          <span
            class="collapsible__title bluedot-h4"
          >
            What is AI Safety?
          </span>
          <span
            class="collapsible__button flex items-center"
          >
            <svg
              class="collapsible__button-icon size-6 transition-transform group-open:rotate-180"
              fill="none"
              stroke="currentColor"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 9l-7 7-7-7"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
          </span>
        </summary>
        <div
          class="collapsible__content pb-6"
        >
          <p
            class="bluedot-p"
          >
            AI Safety is the field working to ensure powerful AI systems benefit rather than harm humanity. As AI systems become more capable of affecting our world, we need to solve key challenges: How do we make AI systems reliably do what we want? How do we govern their development? How do we define what "beneficial" means?
          </p>
          <p
            class="bluedot-p"
          >
            You can read more about these 
            <a
              class="bluedot-a"
              data-rac=""
              href="https://aisafetyfundamentals.com/blog/what-is-ai-alignment/"
              tabindex="0"
            >
              different aspects of AI safety here
            </a>
             and the 
            <a
              class="bluedot-a"
              data-rac=""
              href="https://aisafetyfundamentals.com/blog/ai-risks/"
              tabindex="0"
            >
              range of AI risks here
            </a>
            .
          </p>
        </div>
      </details>
      <details
        class="collapsible max-w-max-width border-b border-color-divider py-4 last:border-b-0 group marker:hidden [&_summary::-webkit-details-marker]:hidden faq-section__item"
      >
        <summary
          class="collapsible__header flex justify-between w-full cursor-pointer py-6 text-left"
        >
          <span
            class="collapsible__title bluedot-h4"
          >
            How can I maximise my chances of getting accepted onto the courses?
          </span>
          <span
            class="collapsible__button flex items-center"
          >
            <svg
              class="collapsible__button-icon size-6 transition-transform group-open:rotate-180"
              fill="none"
              stroke="currentColor"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 9l-7 7-7-7"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
          </span>
        </summary>
        <div
          class="collapsible__content pb-6"
        >
          <p
            class="bluedot-p"
          >
            Check you are the 
            <a
              class="bluedot-a"
              data-rac=""
              href="https://bluedot.org/blog/am-i-the-right-fit-for-bluedots-courses/"
              tabindex="0"
            >
              right fit
            </a>
             and read our tips for making a 
            <a
              class="bluedot-a"
              data-rac=""
              href="https://aisafetyfundamentals.com/blog/avoid-governance-application-mistakes/"
              tabindex="0"
            >
              great application.
            </a>
          </p>
        </div>
      </details>
      <details
        class="collapsible max-w-max-width border-b border-color-divider py-4 last:border-b-0 group marker:hidden [&_summary::-webkit-details-marker]:hidden faq-section__item"
      >
        <summary
          class="collapsible__header flex justify-between w-full cursor-pointer py-6 text-left"
        >
          <span
            class="collapsible__title bluedot-h4"
          >
            I’d like to run your course with my work/university/group of friends. Can you help with training and materials?
          </span>
          <span
            class="collapsible__button flex items-center"
          >
            <svg
              class="collapsible__button-icon size-6 transition-transform group-open:rotate-180"
              fill="none"
              stroke="currentColor"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 9l-7 7-7-7"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
          </span>
        </summary>
        <div
          class="collapsible__content pb-6"
        >
          <p
            class="bluedot-p"
          >
            We’d love for you to do this! All our reading and discussion materials are freely available online. You are welcome to run a local version of our course so long as you 
            <a
              class="bluedot-a"
              data-rac=""
              href="https://bluedot.org/running-versions-of-our-courses/"
              tabindex="0"
            >
              follow our guidelines
            </a>
            .
          </p>
          <p
            class="bluedot-p"
          >
            If you want tips for your discussion facilitation, we recommend taking a look through our 
            <a
              class="bluedot-a"
              data-rac=""
              href="https://course.bluedot.org/home/facilitator-training"
              tabindex="0"
            >
              facilitator training course
            </a>
            .
          </p>
        </div>
      </details>
    </div>
  </section>
</div>
`;
